artifact/v3/scripts/executor.cpp

```diff
@@ -222,18 +279,23 @@ Error ScriptRunner::Execute(
 		this->loop_,
 		[this, current_script, end, ignore_error, handler](Error err) {
 			if (err != error::NoError) {
-				if (ignore_error) {
-					return LogErrAndExecuteNext(err, current_script, end, ignore_error, handler);
+				const bool is_script_retry_error =
+					err.code == processes::MakeError(processes::NonZeroExitStatusError, "").code
+					&& this->script_->GetExitStatus() == state_script_retry_exit_code;
+				if (is_script_retry_error) {
+					MaybeSetupRetryTimeoutTimer();
+					return HandleScriptRetry(current_script, end, ignore_error, handler);
+				} else {
+					if (ignore_error) {
+						return LogErrAndExecuteNext(
+							err, current_script, end, ignore_error, handler);
+					}
+					return HandleScriptError(err, handler);
```

Comment:
You could do `else if ()` and then `return HandleScriptError()` outside of the `if-else if`.